### DISCUSSION TITLE
Contextual Emotes

### DISCUSSION THREAD

**Discussion by revelsix** [SevenTV/Extension#450] [Ideas]
You've got static emotes. You've got animated emotes. You've even got 0-width emotes! Now it's time for the next evolution of emote technology: contextual emotes.

What are contextual emotes? They are a term that I made up for emotes that change orientation and/or color when placed next to another emote. Need an easy example? Let's look at `Clap`. In its current form, it is only capable of clapping to the right, meaning you needed to add a second emote, `Clap2`, in order to do the opposite. For this example, what I'm proposing, and what has been implemented already in [chat.destiny.gg](https://chat.destiny.gg) ([GitHub](https://github.com/destinygg)), would be to automatically flip the emote when in front of another emote.
![image](https://user-images.githubusercontent.com/45189851/229641805-22e5eceb-d06e-4669-bf7c-a2261cfcb35e.png)

d.gg is able to utilize this technology to a much greater extent with their BLADE emote. Not only does it flip orientation depending on the context, but it also changes color depending on your subscriber tier! (It used to change color based on the emote next to it, but that was replaced with this new feature.)
![image](https://user-images.githubusercontent.com/45189851/229642624-de53a304-50f3-4148-8ffd-f2d1e831cb11.png)
![image](https://user-images.githubusercontent.com/45189851/229642871-a2534804-b728-4d26-aa57-b233e01e9efb.png)

Adding this feature would create tons of new opportunities for emote creators to express themselves and would reduce bloat by getting rid of some duplicate emotes.

How would this be implemented? Uh... ***Leaves chat***

**Comment by Treadspear** [comment]
vouch
